src/kv/tx.h

```diff
@@ -74,13 +80,101 @@ namespace kv
       return std::make_tuple(typed_view);
     }
 
+    template <class M>
+    std::tuple<typename M::TxView*> get_tuple2(const std::string& map_name)
+    {
+      if (store == nullptr)
+      {
+        throw std::logic_error("New form called on old-style Tx");
+      }
+
+      using MapView = typename M::TxView;
+
+      // If the M is present, its AbstractTxView should be an M::TxView. This
+      // invariant could be broken by set_view_list, which will produce an error
+      // here
+      auto search = view_list.find(map_name);
+      if (search != view_list.end())
+      {
+        auto view = dynamic_cast<MapView*>(search->second.view.get());
+
+        if (view == nullptr)
+        {
+          throw std::logic_error(
+            fmt::format("View over map {} is not of expected type", map_name));
+        }
+
+        return std::make_tuple(view);
+      }
+
+      if (read_version == NoVersion)
+      {
+        // Grab opacity version that all Maps should be queried at.
+        auto txid = store->current_txid();
+        term = txid.term;
+        read_version = txid.version;
+      }
+
+      M* typed_map = nullptr;
+
+      auto type_erased_map = store->get_map(read_version, map_name);
+      if (type_erased_map == nullptr)
+      {
+        // Store doesn't know this map yet - create it dynamically
+        {
+          const auto map_it = created_maps.find(map_name);
+          if (map_it != created_maps.end())
+          {
+            throw std::logic_error("Created map without creating view over it");
+          }
+        }
+        // TODO: Currently assuming all dynamic maps are replicated and private
+        const bool replicated = true;
+
+        auto new_map = std::make_shared<M>(
```

Comment:
I am sure there's a very good reason but why not `make_unique`?

Reply:
We need to transfer ownership of these maps from this `Tx` to the `Store`, on commit - for that they could be `unique_ptr`. But there's an additional case we need to handle (that we don't currently) - if a transaction is executing over a dynamic map, the map could be rolled back (and deleted) while a `View` over it still exists. Currently this is just a standard pointer - _maybe_ that's fine because we check the rollback counter before accessing the pointer? But to be sure I'd like that `View` to hold a `weak_ptr` or `shared_ptr` to this map, so this map would need to be owned by a `shared_ptr`.